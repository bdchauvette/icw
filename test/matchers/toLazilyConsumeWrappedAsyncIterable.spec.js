test("passes when received value lazily consumes wrapped async iterable", async () => {
  expect.assertions(1);

  await expect(_ => lazyWrapper(_)).toLazilyConsumeWrappedAsyncIterable();

  async function* lazyWrapper(iterable) {
    for await (let value of iterable) yield value;
  }
});

test("fails when received value eagerly consumes wrapped async iterable", async () => {
  expect.assertions(2);

  await expect(
    expect(_ => eagerWrapper(_)).toLazilyConsumeWrappedAsyncIterable()
  ).rejects.toThrowErrorMatchingSnapshot(); // eslint-disable-line jest/prefer-inline-snapshots

  function eagerWrapper(iterable) {
    let iterator = iterable[Symbol.asyncIterator]();
    iterator.next();
    return (async function*() {})();
  }
});

test("fails when received value does not consume wrapped async iterable", async () => {
  expect.assertions(2);

  await expect(
    expect(_ => noopWrapper(_)).toLazilyConsumeWrappedAsyncIterable()
  ).rejects.toThrowErrorMatchingSnapshot(); // eslint-disable-line jest/prefer-inline-snapshots

  async function* noopWrapper() {}
});
