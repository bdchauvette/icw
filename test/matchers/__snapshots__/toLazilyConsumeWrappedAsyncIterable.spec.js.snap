// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fails when received value does not consume wrapped async iterable 1`] = `"expected wrapper to consume wrapped async iterable"`;

exports[`fails when received value eagerly consumes wrapped async iterable 1`] = `"expected wrapper not to eagerly consume wrapped async iterable"`;
